apps/src/netsim/NetSimPacketEditor.js

```diff
@@ -668,6 +691,177 @@ NetSimPacketEditor.prototype.bindElements_ = function () {
 };
 
 /**
+ * Special update method called during send animation that changes the editor
+ * display to show each field left-truncated at an appropriate amount for the
+ * simulated send progress.
+ *
+ * This works differently for different fields:
+ *  - Binary and A/B fields send a single bit at a time.
+ *  - Hex sends a single hex digit at a time, but at the correct slower rate.
+ *  - Decimal and ASCII send one chunk at a time, which depends on the current
+ *    chunk size, and is adjusted to the correct slower rate as well.  For
+ *    ASCII this maps to one character at a time.  For decimal, it's one
+ *    whitespace-delimited number.
+ *
+ * This avoids the jumbled effect of reinterpreting nonbinary fields using
+ * misaligned binary, and communicates in a visual way that it takes longer to
+ * send a single character than it does to send a single bit.
+ *
+ * This method is also designed to send the packet header fields in sequence
+ * before sending the packet body.  Body binary is never seen in the header
+ * fields, each field is treated as an independent space.
+ * @private
+ */
+NetSimPacketEditor.prototype.updateForAnimation_ = function () {
+  var chunkSize = this.currentChunkSize_;
+  var liveFields = [];
+
+  // There may be potential for performance optimization here, but it's not
+  // particularly high on our perf list right now.
+
+  var level = NetSimGlobals.getLevelConfig();
+  var encoder = new Packet.Encoder(level.addressFormat,
+      level.packetCountBitWidth, this.packetSpec_);
+
+  var fieldStart = 0;
+
+  this.packetSpec_.forEach(function (fieldSpec) {
+    /** @type {Packet.HeaderType} */
+    var fieldName = fieldSpec;
+    /** @type {number} */
+    var fieldWidth = encoder.getFieldBitWidth(fieldName);
+
+    if (this.sendAnimationIndex_ < fieldStart + fieldWidth) {
+      // Either we haven't reached this field yet or we're currently animating
+      // through it; don't do anything to the more complex fields, and animate
+      // the binary appropriately.
+      var fieldBinary = this.originalBinary_.substr(fieldStart, fieldWidth);
+      var truncatedBits = Math.max(0, this.sendAnimationIndex_ - fieldStart);
+
+      if (this.isEncodingEnabled_(EncodingType.A_AND_B)) {
+        liveFields.push({
+          inputElement: this.a_and_bUI[fieldName],
```

Comment:
Hmm...That makes it kind of difficult to statically follow/understand the code. Would it be difficult to instead do something like
`this.UI['a_and_b']` instead of `this.a_and_bUI`. I think that would make this easier to follow.